api/utils/requestProcessor.js

```diff
@@ -1922,34 +1930,65 @@ const processRequest = (params) => {
                 case 'download': {
                     if (paths[4] && paths[4] !== '') {
                         common.db.collection("long_tasks").findOne({_id: paths[4]}, function(err, data) {
+                            if (err) {
+                                common.returnMessage(params, 400, err);
+                            }
+                            else {
+                                var filename = data.report_name;
+                                var type = filename.split(".");
+                                type = type[type.length - 1];
+                                var myfile = paths[4];
 
-                            var filename = data.report_name;
-                            var type = filename.split(".");
-                            type = type[type.length - 1];
-                            var myfile = paths[4];
-
-                            countlyFs.gridfs.getSize("task_results", myfile, {id: paths[4]}, function(error, size) {
-                                if (error) {
-                                    common.returnMessage(params, 400, error);
-                                }
-                                else if (parseInt(size) === 0) {
-                                    common.returnMessage(params, 400, "Export size is 0");
-                                }
-                                else {
-                                    countlyFs.gridfs.getStream("task_results", myfile, {id: paths[4]}, function(err5, stream) {
-                                        if (err5) {
-                                            common.returnMessage(params, 400, "Export strem does not exist");
+                                countlyFs.gridfs.getSize("task_results", myfile, {id: paths[4]}, function(err2, size) {
+                                    if (err2) {
+                                        common.returnMessage(params, 400, err2);
+                                    }
+                                    else if (parseInt(size) === 0) {
+                                        if (data.type !== "dbviewer") {
+                                            common.returnMessage(params, 400, "Export size is 0");
                                         }
-                                        else {
-                                            var headers = {};
-                                            headers["Content-Type"] = countlyApi.data.exports.getType(type);
-                                            headers["Content-Disposition"] = "attachment;filename=" + encodeURIComponent(filename);
-                                            params.res.writeHead(200, headers);
-                                            stream.pipe(params.res);
+                                        //handling older aggregations aren't saved in countly_fs
+                                        else if (!data.gridfs && data.data) {
```

Comment:
As I understand this is for case when data is not stored in gridfs(But in report itself)
Lets do it simpler.

There is no need to store it in gridfs.
Please just output data from report in this case.

We are storing in gridfs to make sure it fits.(does not exceed document size) If it is stored in report document - it is already not exceeding. So it is fine for historical ones to stay like that.